cmd/main.go

```diff
@@ -0,0 +1,211 @@
+		}
+
+		// tar utility is available from Windows build 17063
+		// TODO: consider using other command or a custom implementation
+		// https://learn.microsoft.com/en-us/virtualization/community/team-blog/2017/20171219-tar-and-curl-come-to-windows
+		cmd := exec.Command("tar", "-xf", assetDownloadPath, "-C", installDir)
+
+		if out, err := cmd.CombinedOutput(); err != nil {
+			log.Println(string(out))
+			return err
+		}
+
+		return nil
+	default:
+		return errorPlatformNotSupported
+	}
+}
+
+func downloadLatestReleaseAsset() string {
+	release, error := getLatestRelease()
+
+	if error != nil {
+		panic(error)
+	}
+
+	resp, _ := http.Get(*release.asset.BrowserDownloadURL)
+	defer resp.Body.Close()
+
+	out, _ := os.Create(*release.asset.Name)
+	defer out.Close()
+
+	log.Printf("Downloading latest Flowser release %s", release.version)
+
+	_, err := io.Copy(out, resp.Body)
+	if err != nil {
+		panic("Install failed")
+	}
+
+	return out.Name()
+}
+
+type FlowserRelease struct {
+	version string
+	asset   github.ReleaseAsset
+}
+
+func getLatestRelease() (FlowserRelease, error) {
+	client := github.NewClient(nil)
+	release, _, err := client.Repositories.GetLatestRelease(context.Background(), "onflowser", "flowser")
+	if err != nil {
+		log.Fatal(err)
+	}
+	latestVersion := strings.Replace(*release.TagName, "v", "", 1)
+	targetAssetName, err := getAssetName(latestVersion)
+	if err != nil {
+		return FlowserRelease{}, err
+	}
+	for _, asset := range release.Assets {
+		if *asset.Name == targetAssetName {
+			return FlowserRelease{
+				version: latestVersion,
+				asset:   asset,
+			}, nil
+		}
+	}
+	return FlowserRelease{}, errors.New("no asset found")
+}
+
+func getAssetName(version string) (string, error) {
+	isArm := strings.HasPrefix(runtime.GOARCH, "arm")
+	switch runtime.GOOS {
+	case "darwin":
+		if isArm {
+			return fmt.Sprintf("Flowser-%s-arm64-mac.zip", version), nil
+		} else {
+			return fmt.Sprintf("Flowser-%s-mac.zip", version), nil
+		}
+	case "windows":
+		return fmt.Sprintf("Flowser-%s-win.zip", version), nil
+	default:
+		return "", errorPlatformNotSupported
+	}
+}
+
+func getInstallDir() (string, error) {
+	switch runtime.GOOS {
+	case "darwin":
+		return "/Applications/Flowser.app", nil
+	case "windows":
+		// TODO: Search in common install directories
+		// https://superuser.com/questions/1327037/what-choices-do-i-have-about-where-to-install-software-on-windows-10
+		user, err := user.Current()
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("%s\\AppData\\Local\\Programs\\Flowser", user.HomeDir), nil
+	default:
+		return "", errorPlatformNotSupported
+	}
+}
+
+func getExecutableFile() (string, error) {
+	installDir, err := getInstallDir()
+	if err != nil {
+		return "", err
+	}
+	switch runtime.GOOS {
+	case "darwin":
+		return path.Join(installDir, "Contents/MacOS/Flowser"), nil
+	case "windows":
+		return path.Join(installDir, "Flowser.exe"), nil
+	default:
+		return "", errorPlatformNotSupported
+	}
```

Comment:
Anytime you have mappings of values you can use a map :) so what about (a bit pseudo code)
```suggestion
files := map[string]string{ "darwin": "Contents/MacOsx...", "windows": "Flowser.exe" }

file, ok := files[runtime.GOOS]
if !ok {
     return errors.New("platform not supported")
}
return path.Join(installDir, file)
```
Btw also I feel you should get `runtime.GOOS` somewhere at the top and then just pass the value in the functions so you don't have to always check for that value and have the platform not supported errors checks.